src/main/java/org/opentripplanner/datastore/file/ZipFileDataSource.java

```diff
@@ -93,6 +96,27 @@ private void loadContent() {
         ZipEntry entry = entries.nextElement();
         content.add(new ZipFileEntryDataSource(this, entry));
       }
+    } catch (ZipException ze) {
```

Comment:
How long does it take for this exception to be thrown?

Reply:
My understanding is that the whole directory of the ZIP archive is loaded when the first entry or the collection is accessed. On the following accesses contentLoaded is true and the cache directory is used.

So its straight away (untested, yet)

Reply:
I can confirm that the directory is read before the content of the files(if I remember correctly we access the files in random order, not sequential).